### PR TITLE
add common.remoteAuthority

### PR DESCRIPTION
**Pull request by sbatten**

### CODE REVIEW

File: src/vs/platform/telemetry/node/workbenchCommonProperties.ts [hunk 1]
```diff
@@ -29,6 +29,23 @@ export async function resolveWorkbenchCommonProperties(storageService: IStorageS
...
+
+function cleanRemoteAuthority(remoteAuthority?: string): string {
+	if (!remoteAuthority) {
+		return 'none';
+	}
+
+	// Whitelisted remote authorities
+	['ssh-remote', 'dev-container', 'wsl'].forEach((res: string): string | void => {
+		if (remoteAuthority!.indexOf(res) === 0) {
+			return res;
```
Comment (roblourens) [inline src/vs/platform/telemetry/node/workbenchCommonProperties.ts]: You can't return out of a foreach like that

Comment (rebornix) [reply src/vs/platform/telemetry/node/workbenchCommonProperties.ts]: 👍

File: src/vs/platform/telemetry/node/workbenchCommonProperties.ts [hunk 2]
```diff
@@ -11,7 +11,7 @@ export const currentSessionDateStorageKey = 'telemetry.currentSessionDate';
 export const firstSessionDateStorageKey = 'telemetry.firstSessionDate';
 export const lastSessionDateStorageKey = 'telemetry.lastSessionDate';
 
-export async function resolveWorkbenchCommonProperties(storageService: IStorageService, commit: string | undefined, version: string | undefined, machineId: string, installSourcePath: string): Promise<{ [name: string]: string | undefined }> {
+export async function resolveWorkbenchCommonProperties(storageService: IStorageService, commit: string | undefined, version: string | undefined, machineId: string, installSourcePath: string, remoteAuthority?: string, product?: string): Promise<{ [name: string]: string | undefined }> {
```
Comment (rebornix) [inline src/vs/platform/telemetry/node/workbenchCommonProperties.ts]: `product?: string` is not actually used here.

File: src/vs/platform/telemetry/node/workbenchCommonProperties.ts (hunk 1 above)
Comment (rebornix) [reply src/vs/platform/telemetry/node/workbenchCommonProperties.ts]: stupid question here, should we validate `ssh-remote+` instead? Not sure if someone will use authority like `ssh-remote-something+blabla`.

**Review by rebornix — APPROVED** [review]
✅

**Review by roblourens — APPROVED** [review]

**Pull request merged**